[mapdb] Index out of bounds error

## Current Behavior
Running OH3  build #1972 i get the following error on system start up. Due to the fact that every item is persisted with mapdb and i have currently more that 500 item, i did not see any missing data.

```
2020-10-18 13:40:12.958 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'QueryablePersistenceService.query()' on 'org.openhab.persistence.mapdb.internal.MapDbPersistenceService@710e5de1': Index 1 out of bounds for length 1
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:48) ~[?:?]
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:1) ~[?:?]
        at com.google.gson.internal.bind.TypeAdapters$35$1.read(TypeAdapters.java:893) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:888) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:853) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:802) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:774) ~[bundleFile:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.deserialize(MapDbPersistenceService.java:168) ~[?:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.query(MapDbPersistenceService.java:155) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor42.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-18 13:40:12.984 [ERROR] [ence.internal.PersistenceManagerImpl] - Exception occurred while querying persistence service 'mapdb': Index 1 out of bounds for length 1
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:48) ~[?:?]
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:1) ~[?:?]
        at com.google.gson.internal.bind.TypeAdapters$35$1.read(TypeAdapters.java:893) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:888) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:853) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:802) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:774) ~[bundleFile:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.deserialize(MapDbPersistenceService.java:168) ~[?:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.query(MapDbPersistenceService.java:155) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor42.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-18 13:40:15.192 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'QueryablePersistenceService.query()' on 'org.openhab.persistence.mapdb.internal.MapDbPersistenceService@710e5de1': Index 1 out of bounds for length 1
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:48) ~[?:?]
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:1) ~[?:?]
        at com.google.gson.internal.bind.TypeAdapters$35$1.read(TypeAdapters.java:893) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:888) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:853) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:802) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:774) ~[bundleFile:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.deserialize(MapDbPersistenceService.java:168) ~[?:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.query(MapDbPersistenceService.java:155) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor42.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-18 13:40:15.199 [ERROR] [ence.internal.PersistenceManagerImpl] - Exception occurred while querying persistence service 'mapdb': Index 1 out of bounds for length 1
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:48) ~[?:?]
        at org.openhab.persistence.mapdb.internal.StateTypeAdapter.read(StateTypeAdapter.java:1) ~[?:?]
        at com.google.gson.internal.bind.TypeAdapters$35$1.read(TypeAdapters.java:893) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:129) ~[bundleFile:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:220) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:888) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:853) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:802) ~[bundleFile:?]
        at com.google.gson.Gson.fromJson(Gson.java:774) ~[bundleFile:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.deserialize(MapDbPersistenceService.java:168) ~[?:?]
        at org.openhab.persistence.mapdb.internal.MapDbPersistenceService.query(MapDbPersistenceService.java:155) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor42.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

## Your Environment
openhab Version: 3 build 1972
HW: Virtual maschine on Synology NAS
OS: Debian 10

--- a/bundles/org.openhab.persistence.mapdb/src/main/java/org/openhab/persistence/mapdb/internal/StateTypeAdapter.java
+++ b/bundles/org.openhab.persistence.mapdb/src/main/java/org/openhab/persistence/mapdb/internal/StateTypeAdapter.java
@@ -43,11 +43,12 @@ public class StateTypeAdapter extends TypeAdapter<State> {
             return null;
         }
         String value = reader.nextString();
-        String[] parts = value.split(TYPE_SEPARATOR);
-        String valueTypeName = parts[0];
-        String valueAsString = parts[1];
 
         try {
+            String[] parts = value.split(TYPE_SEPARATOR);
+            String valueTypeName = parts[0];
+            String valueAsString = parts[1];
+
             @SuppressWarnings("unchecked")
             Class<? extends State> valueType = (Class<? extends State>) Class.forName(valueTypeName);
             List<Class<? extends State>> types = Collections.singletonList(valueType);

